atlasdb-impl-shared/src/main/java/com/palantir/atlasdb/sweep/SpecificTableSweeper.java

```diff
@@ -201,12 +199,19 @@ private static SweepResults getCumulativeSweepResults(TableToSweep tableToSweep,
         long minimumSweptTimestamp = Math.min(
                 tableToSweep.getPreviousMinimumSweptTimestamp().orElse(Long.MAX_VALUE),
                 currentIteration.getSweptTimestamp());
+        long timeInMillis = tableToSweep.getTimeInMillisPreviously() + currentIteration.getTimeInMillis();
+        // todo(gmaretic): should we use some other Clock?
+        long timeElapsedSinceStart = tableToSweep.getStartTimeInMillis()
+                .map(start -> new SystemClock().getTimeMillis() - start)
+                .orElse(currentIteration.getTimeInMillis());
 
         return SweepResults.builder()
```

Comment:
I don't like how we're using SweepResults - it seems to be partially instantiated in a number of places.  The only way you know what is safe to call on a particular instance is to read back through the code to find out what has been set on the instance you've got which seems dodgy to me.

We should either make sure we copy all values across from one instantiation to another (i.e. when making the cumulative metrics), abandon Immutables here and force devs to use a constructor, or have separate classes for the different types of SweepResults.